apps/backoffice/src/components/services/subscriptions-migration/migration-manager.tsx

```diff
@@ -0,0 +1,205 @@
+import { getConfiguration } from "@/config";
+import { MigrationData } from "@/generated/api/MigrationData";
+import { MigrationDelegate } from "@/generated/api/MigrationDelegate";
+import { MigrationDelegateList } from "@/generated/api/MigrationDelegateList";
+import { MigrationItemList } from "@/generated/api/MigrationItemList";
+import useFetch from "@/hooks/use-fetch";
+import { Alert, AlertTitle, Divider, Stack, Typography } from "@mui/material";
+import axios from "axios";
+import { useTranslation } from "next-i18next";
+import { enqueueSnackbar } from "notistack";
+import { useEffect, useState } from "react";
+import { MigrationLatest } from ".";
+import { MigrationModal } from "./migration-modal";
+
+// migration status as it matters for this component
+export type MigrationStatus = "done" | "doing" | "todo" | "failed";
+export type MigrationStatusColor = "info" | "success" | "warning" | "error";
+export type MigrationChipStatus = {
+  label: MigrationStatus;
+  color: MigrationStatusColor;
+};
+
+export type DelegateStatusPair = {
+  delegate: MigrationDelegate;
+  data: MigrationData;
+};
+
+/** Renders subscriptions migrator helper _(from old Deleloper Portal)_ */
+export const MigrationManager = () => {
+  const { t } = useTranslation();
+  const [isImportModalOpen, setIsImportModalOpen] = useState(false);
+  const [importInProgress, setImportInProgress] = useState(false);
+  const [migrationStatusList, setMigrationStatusList] = useState<
+    Array<DelegateStatusPair>
+  >();
+
+  const {
+    data: migrationItemsData,
+    loading: migrationItemsLoading,
+    fetchData: migrationItemsFetchData
+  } = useFetch<MigrationItemList>();
+  const {
+    data: migrationDelegatesData,
+    loading: migrationDelegatesLoading,
+    fetchData: migrationDelegatesFetchData
+  } = useFetch<MigrationDelegateList>();
+
+  const handleOpenImportModal = () => {
+    setIsImportModalOpen(true);
+  };
+
+  const getMigrationOwnershipClaimsUrl = (delegateId: string) =>
+    `${
+      getConfiguration().API_BACKEND_BASE_URL
+    }/api/services/migrations/ownership-claims/${delegateId}`;
+
+  // direct axios fetch
+  const getDelegateMigrationStatus = async (delegateId: string) => {
+    try {
+      const { data, status } = await axios.get<MigrationData>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { data, status };
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        return { data: error.message, status: -1 };
+      } else {
+        return { data: JSON.stringify(error), status: -1 };
+      }
+    }
+  };
+
+  // direct axios fetch
+  const postDelegateMigrationOwnershipClaims = async (delegateId: string) => {
+    try {
+      const { status } = await axios.post<void>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { status };
+    } catch (error) {
+      return { status: -1 };
+    }
+  };
+
+  const fetchDelegateStatusPairs = () => {
```

Comment:
I'd make this method take `migrationDelegatesData` as input, see comment on useEffect

Reply:
`migrationDelegatesData` is the result of `useFetch` custom hook and for this reason I cannot use it "explicitly", at least in this context.
**P.S.:** Since I have an upcoming release deadline, I'm adding this issue as a technical debt task that I will evaluate asap.
Thanks for suggestions :-)